src/dirents_to_file_entries.ts

```diff
@@ -14,7 +14,7 @@ export function* direntsToFileEntries<P extends Path<string>>(
   files: Dirent[],
   root: P,
   filter: (file: Dirent) => boolean,
-): IterableIterator<FileEntry<`${P}/${string}`>> {
+): Generator<FileEntry<`${P}/${string}`>> {
```

Comment:
The correct type is `IteratorObject` if you want to avoid implementation detail that we return a `Generator`.

Refs: https://2ality.com/2025/06/typescript-iterator-types.html#typescript%3A-preparing-for-the-new-methods